src/Block/Loader/ServiceLoader.php

```diff
@@ -35,19 +36,22 @@ public function __construct(array $types)
      * Check if a given block type exists.
      *
      * @param string $type Block type to check for
-     *
-     * @return bool
      */
-    public function exists($type)
+    public function exists(string $type): bool
     {
         return \in_array($type, $this->types, true);
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function load($configuration)
+    public function load($configuration): BlockInterface
     {
+        if (!\is_string($configuration) && !\is_array($configuration)) {
```

Comment:
Duplicate code. Can't you just call the `support` method?

Reply:
I didn't perform that simplification in order to not change the call flow (`load()` and `support()`). If it doesn't worth, I can make the update as suggested.